### PR TITLE
Faceted search mock

### PR DESCRIPTION
**Pull request by kewitham**
image

### CODE REVIEW

File: assets/js/pagination.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { NUMBER_OF_RESULTS, SEARCH_ENDPOINT, ACCESS_KEY, AFFILIATE } from './utils/constants';
 import renderSearchPage from './utils/renderSearchPage';
 import totalResults from './templates/search/totalResultsTemplate';
+import { _ } from 'core-js';
```
Comment (okkays) [inline assets/js/pagination.js:4]: can you add a comment to explain what this is polyfilling?

File: assets/js/pagination.js [hunk 2]
```diff
@@ -11,10 +12,20 @@ let searchParams = {
   query: urlParams.get('query'),
   offset: urlParams.get('offset') || 0,
   limit: NUMBER_OF_RESULTS,
+ // audience: urlParams.get('audience'),
```
Comment (okkays) [inline assets/js/pagination.js:15]: (remove commented out code)

File: assets/js/pagination.js [hunk 3]
```diff
@@ -11,10 +12,20 @@ let searchParams = {
...
 const target = document.getElementById('totalResultsTarget');
-
+const hiddenQueryInput = document.getElementById('hiddenQuery');
+hiddenQueryInput.value = urlParams.get('query');
+document.querySelectorAll('.usa-checkbox__input').forEach((checkbox) => {
+  if (checkbox.value == audience) {
+    checkbox.setAttribute('data-checked', true);
+  } else {
+    checkbox.setAttribute('data-checked', false);
+  }
```
Comment (okkays) [inline assets/js/pagination.js:27]:
```suggestion
checkbox.setAttribute('data-checked', checkbox.value === audience);
```

File: assets/js/utils/renderSearchPage.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ import searchResultsTemplate from '../templates/search/searchResultsTemplate';
 import totalResults from '../templates/search/totalResultsTemplate';
 import clickTracking from '../click-tracking';
 
-export default function renderSearchPage(searchResults, urlParams, numberOfResults) {
+export default function renderSearchPage(searchResults, urlParams, numberOfResults, audience = null) {
```
Comment (okkays) [inline assets/js/utils/renderSearchPage.js:15]: You should be able to do this without the default parameter - in javascript, by default, parameters that aren't specified will be `undefined`

```suggestion
export default function renderSearchPage(searchResults, urlParams, numberOfResults, audience) {
```

File: assets/js/utils/renderSearchPage.js [hunk 5]
```diff
@@ -48,8 +48,11 @@ export default function renderSearchPage(searchResults, urlParams, numberOfResul
     const pagination_list = document.querySelectorAll('ol.usa-pagination__list')[0];
     // Put the fetched results into a list, and render in the DOM:
     webResults.forEach(function (item) {
-      renderSearchResults(searchResultsTemplate(item));
-    });
+      if (!audience || item.url.includes(audience)) {
+        renderSearchResults(searchResultsTemplate(item));
+      }
+      return null
```
Comment (okkays) [inline assets/js/utils/renderSearchPage.js:54]:
```suggestion
      return null;
```

**Review by okkays — CHANGES_REQUESTED** [review]
code looking good, some style comments

**Review by okkays — CHANGES_REQUESTED** [review]
Missed something, sorry!

**Comment by vwilliamsn** [comment]
@kewitham great job! Quick question about facet search, I tried searching for "door & doors" using the Topics filter and it only surfaces the 'Voting' overlay. Door(s) also appears in the State and Local Government & Businesses That Are Open to the Public however it's behind the accordion feature. Can the search pull from something that’s hidden behind the accordion feature?

**Comment by kewitham** [comment]
> @kewitham great job! Quick question about facet search, I tried searching for "door & doors" using the Topics filter and it only surfaces the 'Voting' overlay. Door(s) also appears in the State and Local Government & Businesses That Are Open to the Public however it's behind the accordion feature. Can the search pull from something that’s hidden behind the accordion feature?

@vwilliamsn the filter won't change which results are surfaced for certain keywords and I don't see that result coming up on the regular search currently on the site. It should pull it even though it's in an accordion though. Maybe we can investigate updating the meta data to include that keyword. Do you want me to create a ticket for that?

**Comment by sanchez-amanda** [comment]
A few style thoughts that i put in this screenshot - mostly moving headers around to match the style of guidance and resources page.

- Move 'Search' header to top of results column (like guidance and resources page)
- Once header is moved, move ‘document type’ so that the top of the text lines up with ‘Search’ - similar to Guidance and Resource Materials
- Open Question: did we want to add descriptors to the document types? tooltip? not necessary for approval but just curious where we landed with that

Screenshot 2024-02-13 at 11 15 20 AM

@kewitham @vwilliamsn

**Comment by vwilliamsn** [comment]
@sanchez-amanda - love that suggestion. I think having descriptors or a tooltip would give folks more of an understanding of the new filter options and how that might improve their search results. Do you have a recommendation - should we use the info box brief description or a more detail description of the different content types?

@kewitham - on the left side, we list the content types, can we mirror the global navigation wording. What's the lift for adding descriptors or a tooltip feature?
image

**Comment by kewitham** [comment]
@vwilliamsn @sanchez-amanda sounds good! yes I actually already added a tooltip (for some reason the icon isn't showing up in the federalist preview). Right now the language I have is: "Basic overview" for Topics, "Detailed guidance" for Resources and "Legal documents" for Law and Regulations.

**Comment by kewitham** [comment]
@vwilliamsn @sanchez-amanda made the requested updates!

**Review by okkays — APPROVED** [review]
LGTM!

**Pull request merged**